blocks/library/latest-posts/block.js

```diff
@@ -157,11 +157,11 @@ class LatestPostsBlock extends Component {
 }
 
 export default withAPIData( ( props ) => {
-	const { postsToShow, order, orderBy, categories } = props.attributes;
+	const { postsToShow, order, orderby, categories } = props.attributes;
 	const latestPostsQuery = stringify( pickBy( {
 		categories,
 		order,
-		orderBy,
+		orderby,
```

Comment:
While your solution works with the latest posts block as a quick patch, it may create conflict with the query controls component (select orderby list) if used with the right syntax by another block. In this case, the select list will not match the selected orderby value. In my opinion, it will create to much confusions and it partially fixes the issue with query controls component. It means that all blocks which will use the query controls component must have the same typo in order to correctly work.

Reply:
We are moving away from `withAPIData` approach to fetching data, so this is going to be further moved to become an implementation detail. I don't think that code structure should be enforced by the API internal bits. We should fix this particular case only. `QueryControls` is a general purpose component and doesn't need to be tied to this REST API call.

Reply:
So, you still need to fix the `QueryControls` because it matches a certain syntax which is not sustainable as a general purpose component as mentioned in my previous reply. Destructuring the object in `QueryControls` enforces a certain syntax in blocks. So, or the syntax is coherent between a block and a general component, or the syntax of the general component must not rely on a block syntax (`QueryControls` was created for and with the latest posts block syntax). It's to much confusing in my opinion. This error is not just because of the REST API syntax, that's the main reason that I have changed all related files. Making a quick patch instead of a deep fix will increase the technical debt and make things confusing. It's still possible to make things clear for everyone.

Reply:
@jorgefilipecosta, any thought on this one?

Reply:
Hi @loicblascos,
Thank you for catching this issue and submitting a fix for it 👍 

Regarding the changes being discussed, currently, QueryControls receives two props orderby ( with the current value for orderBy ) and onOrderByChange ( a function called when orderby changes). I think this is not necessarily wrong that's the name of the props the component receives and the users of the component can name the attributes where the prop is saved in any way they want. And this does not necessarily need to map to some field in the rest API.

That said, It looks like WordPress core rest API interprets orderby as a "single word" otherwise it would have called it order_by.  In Gutenberg, we interpret it as two words (orderBy).
I think unifying and making Gutenberg also use orderby may make developers life easier as we are also treating as a "single word" like core.

If we follow this path I think we should also rename prop onOrderByChange to onOrderbyChange and we need to have a migrate function that migrates current latest post blocks to the new attribute name as specified in https://github.com/WordPress/gutenberg/blob/master/docs/deprecated-blocks.md. @loicblascos if you need any help with the migrate function just let me know and I will try to add commit/code with it.

Reply:
Hi @loicblascos,
Unfortunately, I think we can not advance with this approach because we checked that we don't have yet a way to migrate server-side blocks. The migrate in the client just affects blocks when the post is opened in the editor but for existing posts, not open in the editor, the attribute would still be wrong on the server side.
It looks like the way we can resolve this, is what @gziolo suggested and just map the request to orderby in the withAPIData data and leave the attribute name and props unchanged. In the server side function, we would also map the attribute. This is not wrong as the attribute names don't need to relate to the fields in rest API.
I'm really sorry for pointing you in the wrong direction, at the time I was not aware that migrate for server-side blocks still needs to be implemented.
Thank you for your contributions!

Reply:
Hi @jorgefilipecosta,
I just pushed a commit to map the `orderBy` attribute...

Reply:
Hi @loicblascos,
Thank you for the updates.
It looks like some problem happened and this branch diff contains unrelated changes, for example, e.g:className on the index.php.
The branch also contains some conflicts, I think we can handle both problems with a rebase.

Reply:
Hi @jorgefilipecosta,
It should be fine now.

Reply:
HI @loicblascos, the PR was merged. Thank you for all the effort during the iterations!